packages/esm-patient-notes-app/src/notes/visit-notes-form.component.tsx

```diff
@@ -112,42 +137,63 @@ const VisitNotesForm: React.FC<DefaultWorkspaceProps> = ({ closeWorkspace, patie
     [],
   );
 
-  const handleAddDiagnosis = (diagnosisToAdd: Diagnosis) => {
-    setSearchTerm('');
-    setSearchResults(null);
-    setSelectedDiagnoses((selectedDiagnoses) => [...selectedDiagnoses, diagnosisToAdd]);
+  const handleAddDiagnosis = (conceptDiagnosisToAdd: Concept, searchInputField: string) => {
+    let newDiagnosis = createDiagnosis(conceptDiagnosisToAdd);
+    if (searchInputField == 'primaryInputSearch') {
+      newDiagnosis.rank = 1;
+      setPrimarySearchTerm('');
+      setSearchPrimaryResults(null);
+      setSelectedPrimaryDiagnoses((selectedDiagnoses) => [...selectedDiagnoses, newDiagnosis]);
+    } else if (searchInputField == 'secondaryInputSearch') {
+      setSecondarySearchTerm('');
+      setSearchSecondaryResults(null);
+      setSelectedSecondaryDiagnoses((selectedDiagnoses) => [...selectedDiagnoses, newDiagnosis]);
+    }
+    setCombinedDiagnoses((diagnosisCombined) => [...diagnosisCombined, newDiagnosis]);
   };
 
-  const handleRemoveDiagnosis = (diagnosisToRemove: Diagnosis) => {
-    setSelectedDiagnoses(
-      selectedDiagnoses.filter((diagnosis) => diagnosis.concept.id !== diagnosisToRemove.concept.id),
+  const handleRemoveDiagnosis = (diagnosisToRemove: Diagnosis, searchInputField: string) => {
+    if (searchInputField == 'primaryInputSearch') {
+      setSelectedPrimaryDiagnoses(
+        selectedPrimaryDiagnoses.filter((diagnosis) => diagnosis.diagnosis.coded !== diagnosisToRemove.diagnosis.coded),
+      );
+    } else if (searchInputField == 'secondaryInputSearch') {
+      setSelectedSecondaryDiagnoses(
+        selectedSecondaryDiagnoses.filter(
+          (diagnosis) => diagnosis.diagnosis.coded !== diagnosisToRemove.diagnosis.coded,
+        ),
+      );
+    }
+    setCombinedDiagnoses(
+      combinedDiagnoses.filter((diagnosis) => diagnosis.diagnosis.coded !== diagnosisToRemove.diagnosis.coded),
     );
   };
 
+  const createDiagnosis = (concept: Concept) => {
+    return {
+      patient: patientUuid,
+      diagnosis: {
+        coded: concept.uuid,
+      },
+      rank: 2,
+      certainty: 'PROVISIONAL',
+      display: concept.display,
+    };
+  };
+
   const handleSubmit = React.useCallback(
     (event: SyntheticEvent) => {
+      setIsHandlingSubmit(true);
       event.preventDefault();
 
-      if (!selectedDiagnoses.length) return;
-
-      if (selectedDiagnoses.length) {
-        selectedDiagnoses[0].primary = true;
-      }
-
-      let obs = convertToObsPayload(selectedDiagnoses);
-
-      if (clinicalNote) {
-        obs = [
-          {
-            concept: encounterNoteTextConceptUuid,
-            value: clinicalNote,
-          },
-          ...obs,
-        ];
+      if (!selectedPrimaryDiagnoses.length) {
```

Comment:
Admittedly, I'm a little confused about why we bale here if no primary diagnoses are provided? Shouldn't this be `!combinedDiagnoses.length`?

Reply:
This should be because only the primary diagnoses are mandatory.